ts/src/profiler.ts

```diff
@@ -268,25 +330,31 @@ export class Profiler extends common.ServiceObject {
       body: reqBody,
       json: true,
     };
-
+    
     this.logger.debug(`Attempting to create profile.`);
-    const [prof, response] = await this.request(options);
-    if (!hasHttpStatusCode(response)) {
-      throw new Error('Server response missing status information.');
-    }
-    if (isErrorResponseStatusCode(response.statusCode)) {
-      let message: number|string = response.statusCode;
-      // tslint:disable-next-line: no-any
-      if ((response as any).statusMessage) {
-        message = response.statusMessage;
-      }
-      throw new Error(message.toString());
-    }
-    if (!isRequestProfile(prof)) {
-      throw new Error(`Profile not valid: ${JSON.stringify(prof)}.`);
-    }
-    this.logger.debug(`Successfully created profile ${prof.profileType}.`);
-    return prof;
+    return new Promise<RequestProfile>((resolve, reject) => {
+      this.request(
+          options,
+          (err: Error, prof: object, response: http.ServerResponse) => {
+            if (response && isErrorResponseStatusCode(response.statusCode)) {
+              if (isServerBackoffResponse(response)) {
+                throw new BackoffResponseError(response);
+              }
+              throw new Error(response.statusMessage);
+            }
+            if (err) {
+              reject(err);
```

Comment:
You use `reject(err)` here but `throw new SomeError()` a couple lines above - is there a difference?

Reply:
There somewhat similar -- but I've switched to reject, for consistency and because it seems like there are certain cases where thrown errors can be lost.